Lib/asyncio/unix_events.py

```diff
@@ -1266,7 +1266,13 @@ def is_active(self):
         return True
 
     def close(self):
-        pass
+        self._join_threads()
```

Comment:
Hum, it's unclear to me if close() is called in the normal usage of asyncio. It seems like by default, only `__del__()` is called.

Can we imagine to add a new method to only wait for threads spawned by an event loop? Pseudo-code:

```
def _close_loop(self, loop):
  threads = [thread for thread in self._threads.values() if thread.loop == loop]
  self._join_threads(threads)
```

And then call watcher._close_loop(self) in _UnixSelectorEventLoop.close().

Reply:
Responded to this in https://github.com/python/cpython/pull/16552#issuecomment-537892200.

Reply:
@vstinner 
> Can we imagine to add a new method to only wait for threads spawned by an event loop? Pseudo-code:

```
def _close_loop(self, loop):
  threads = [thread for thread in self._threads.values() if thread.loop == loop]
  self._join_threads(threads)
```

Also, I don't think we have a way to determine if the thread was spawned by an event loop, see [Thread's implementation](https://github.com/python/cpython/blob/86ec5c65fe04ece16f3aada5292cc62726062ad0/Lib/threading.py#L757).

However, it might be a good idea to only close the non-daemon threads, instead of all of the ones that are alive. The program can exit if there daemon threads still open.

```python
    def _join_threads(self):
        threads = [thread for thread in list(self._threads.values())
                            if thread.is_alive() and not thread.daemon]
        for thread in threads:
            thread.join()
```